### PR TITLE
Support HTML elements as textures when DOM/canvas are available

### PR DESCRIPTION
**Pull request by Benjamin-Dobell**

### CODE REVIEW

File: webgl.js [hunk 1]
```diff
@@ -3482,6 +3482,36 @@ function checkFormat (format) {
...
+        context = canvas.getContext('2d')
+
+        if (context !== null) {
+          context.drawImage(pixels, 0, 0)
+        }
+      }
+    }
+  }
+
+  if (context !== null) {
```
Comment (gre) [inline webgl.js]: looks like this `if` should be moved inside the previous `if` where the context variable is defined?

Comment (Benjamin-Dobell) [inline webgl.js]: @gre Whoops, quite right. Originally I had the `context` variable declaration outside the outer `if` statement but moved it just before submitting the pull request.

Force pushing incoming...

**Comment by gre** [comment]
@mikolalysenko what do you think of this feature? should we merge this?

I think it's great for more interoperability stuff, I actually need it for some tests :)

File: webgl.js [hunk 2]
```diff
@@ -3482,6 +3482,36 @@ function checkFormat (format) {
...
+var extractImageData = function (pixels) {
+  if (typeof pixels === 'object' && typeof pixels.width !== 'undefined' && typeof pixels.height !== 'undefined') {
+    if (typeof pixels.data !== 'undefined') {
+      return pixels
+    }
+
+    var context = null
+
+    if (typeof pixels.getContext === 'function') {
+      context = canvas.getContext('2d')
```
Comment (Benjamin-Dobell) [inline webgl.js]: Whoops, don't merge yet.

This should be `pixels` not `canvas` on this line.

**Comment by Benjamin-Dobell** [comment]
Alright just force pushed a fix.

**Comment by regou** [comment]
This pr works so will and solve many canvas based texture issues but never get merged. Why?
@substack @mourner 
#45

**Comment by mourner** [comment]
Sorry for not reviewing this sooner. Looks good to me!

**Comment by jhollingworth** [comment]
Could someone publish a new version for this so we don't have to reference git?

**Pull request merged**